scripts/api_diff_report/api_diff_report.py

```diff
@@ -0,0 +1,259 @@
+    for key, value in diff.items():
+        if isinstance(value, dict):
+            if key in ['api_types', 'apis', 'sub_apis']:
+                report += generate_markdown_report(value, level=level)
+            else:
+                current_status = value.get('status')
+                if current_status:
+                    if level == 0:  # Module level: Always print out module name as title
+                        report += f'{header_str} {key} [{current_status}]\n'
+                    if current_status != STATUS_ERROR:  # ADDED,REMOVED,MODIFIED
+                        report += f'<details>\n<summary>\n[{current_status}] {key}\n</summary>\n\n'
+                        declarations = value.get('declaration', [])
+                        sub_report = generate_text_report(value,
+                                                          level=1,
+                                                          print_key=False)
+                        detail = process_declarations(current_status,
+                                                      declarations, sub_report)
+                        report += f'```diff\n{detail}\n```\n\n</details>\n\n'
+                else:  # no diff at current level
+                    report += f'{header_str} {key}\n'
+                    report += generate_markdown_report(value, level=level + 1)
+
+                if level == 0:  # Module level: Always print out divider at the end
+                    report += '-----\n'
+
+    return report
+
+
+def process_declarations(current_status, declarations, sub_report):
+    """Diff syntax highlighting in Github Markdown."""
+    detail = ''
+    if current_status == STATUS_MODIFIED:
+        for line in (declarations + sub_report.split('\n')):
+            if STATUS_ADD in line:
+                prefix = '+ '
+                continue
+            elif STATUS_REMOVED in line:
+                prefix = '- '
+                continue
+            if line:
+                detail += f'{prefix}{line}\n'
+    else:
+        prefix = '+ ' if current_status == STATUS_ADD else '- '
+        for line in (declarations + sub_report.split('\n')):
+            if line:
+                detail += f'{prefix}{line}\n'
+
+    return categorize_declarations(detail)
+
+
+def categorize_declarations(detail):
+    """Categorize API info by Swift and Objective-C."""
+    lines = detail.split('\n')
+
+    swift_lines = [
+        line.replace('Swift', '') for line in lines if 'Swift' in line
+    ]
+    objc_lines = [
+        line.replace('Objective-C', '') for line in lines
+        if 'Objective-C' in line
+    ]
+
+    swift_detail = 'Swift:\n' + '\n'.join(swift_lines) if swift_lines else ''
+    objc_detail = 'Objective-C:\n' + '\n'.join(
+        objc_lines) if objc_lines else ''
+
+    if not swift_detail and not objc_detail:
+        return detail
+    else:
+        return f'{swift_detail}\n{objc_detail}'.strip()
+
+
+def parse_cmdline_args():
```

Comment:
nit: could you add support for `--help` in case someone is running it locally and wants quick docs on how to use?

Reply:
Can this be easily run locally? If not, then disregard above comment.

Reply:
We can run it locally, but not intuitive.
If there is requirement to run it locally. After the project being merged. I can have a ReadMe file for the tool.